### DISCUSSION TITLE
Is coolify supposed to reverse proxy to services deployed on a remote server?

### DISCUSSION THREAD

**Discussion by bmcgonag** [coollabsio/coolify#3567] [General]
I have been testing with Coolify, and it's awesome! Thank you so much for this. 

My question is around deploying through coolify to a remote server.  When I do this, the reverse proxy address never works. It works fine if I deploy to localhost.  Is this supposed to function properly, or am I setting something up incorrectly?

For reference, I'm running coolify behind a firewall inside a private LAN, and so have port 80 and 443 forwarded to the private IP for my coolify instance.   It is not running directly on the public IP as it might if it were installed on a VPS.

Thanks, and willing to help with any additional info or troubleshooting.

**Comment by vkartk** [comment]
@bmcgonag Remote server have there own reverse proxy. 

Please go the remote server -> Proxy
 Select Traefik or Caddy and start the proxy
Make sure you point the domain to the Remote server IP.

**Reply by bmcgonag** [reply]
Ok. That's a shame, because the built-in setup of the reverse proxy is one of the most beneficial features of Coolify IMO.  I, don't run it on the open internet, so each server only has a private IP, not  a public IP.  Thus, when a request comes in from the internet for a URL, I have to proxy that request to the right server (today I use NGinX Proxy Manager to do this), as each service runs on different internal servers with varying private IPs.  

To use Coolify inside a LAN they way you are saying, you'd have to reverse proxy to the coolify servers, and pass along the specifics on each URL coolify creates, which actually increases the work to setup seervices at home if you use multiple sservers. 

Does that make sense?

**Reply by vkartk** [reply]
@bmcgonag Why not run a single VM with the servers clustered together? Also, you don't need to make the Coolify instance public unless you plan to access it via WAN or perform API calls.

If you have multiple servers with private IPs, you can use Cloudflare tunnels as an alternative.

**Comment by freezy** [comment]
I had the same problem and ended up building a reverse proxy Nginx image for the ingress.

But I agree, automatic ingress handling for private sub networks would be nice.
